### ISSUE TITLE
Improve PR Check workflow so that it builds the extension locally, pushes the image into local registry and it is then used in E2E tests

### ISSUE THREAD

**Issue by odockal** [containers/podman-desktop-extension-bootc#330]
### Is your enhancement related to a problem? Please describe

PRcheck workflow with e2e tests builds local extension in a way, it builds an image, creates a container and then exports the container into an archive which is extracted into location that is used for testing podman-desktop instance.

### Describe the solution you'd like

It would be best to create local registry, build the image and push the image into that registry. Image from local registry then can be installed in e2e tests via OCI image that uses registry protocol.

**Comment by cdrage** [comment]
@odockal Just going through some old issues, could we close this now that testing covers it?

**Comment by odockal** [comment]
@cdrage I do not think this was addressed. It is how we build the extension image on PR check workflow.
